### ISSUE TITLE
Combined list TLD's

### ISSUE THREAD

**Issue by devipasigner** [hagezi/dns-blocklists#143]
Missing Abused TLDs from yokoffing nextdns

.agency
.ci
.fun
.link
.live
.shop
.win

Thank you, I have some other personal abused TLDs, I will send them for review once I am home

**Comment by devipasigner** [comment]
.lol
.one
.online

.email
.sex
.sexy
.recipes
.xxx
.yandex
.zone
.software

**Comment by bestplayerbot** [comment]
wouldn't .xxx have too many false positives?

**Comment by hagezi** [comment]
Thank you, I will check later which ones lead to many false positives and which ones do not.

*.xxx is probably more for the personal block list.

**Comment by hagezi** [comment]
Result of the occurrence of the TLDs on the Umbrella Toplist (contains no malicious doamins):

TLD | on toplist
-- | --:
sexy | 3
recipes | 8
yandex | 13
ci | 20
sex | 35
agency | 50
software | 54
win | 157
email | 163
xxx | 177
lol | 220
zone | 223
shop | 338
fun | 351
one | 558
live | 904
online | 981
link | 1241

I think `one, live, online, and link` should not be added.

@devipasigner @bestplayerbot: What do you think?

**Comment by hagezi** [comment]
Have already added the "safe" ones:

|*.sexy^
|*.recipes^
|*.ci^
|*.sex^
|*.agency^
|*.software^
|*.win^
|*.email^

**Comment by devipasigner** [comment]
> 

almost all but 2 of the tlds excluded are from  @yokoffing
I haven't experienced any false positives and it has been semi effective but maybe it would be best to get an answer from @yokoffing

**Comment by bestplayerbot** [comment]
I've seen some false positives with  *.link

**Comment by yokoffing** [comment]
> Result of the occurrence of the TLDs on the Umbrella Toplist

This is helpful! Thank you.

**Comment by yokoffing** [comment]
> `one, live, online, and link` should not be added.

@hagezi Spamhaus says `one` is 3.6% bad and `online` 2.6% bad, whereas `live` is 25% bad. `link` is 12.5%. (For reference, `.com` is 1.7%.)

- I think we can drop `one` and `online` from hardblocking at the DNS-level.
- would add `live`
- I'd leave `link` out of hardblocking but may leave on a filterlst to warn the user. But I favor towards removal: it's starting to be used more.

**Comment by yokoffing** [comment]
@devipasigner 

> .email
> .sex
> .sexy
> .recipes
> .xxx
> .yandex
> .zone
> .software

Spamhaus is not everything, but one tool to take into consideration:

|  **TLD** | **Spamhaus % Bad Domains** |
|:--------:|:--------------------------:|
|  com (reference)   |            1.7%         |
|  org (reference)   |            1.2%         |
|   email  |             4%             |
|    sex   |             0%             |
|   sexy   |             0%             |
|  recipes |             0%             |
|    xxx   |             0%             |
|  yandex  |             0%             |
|   zone   |             11.1%            |
| software |            5.8%            |

**Comment by yokoffing** [comment]
I went through my filterlist and purged TLDs that are less than 10% bad according to Spamhaus https://github.com/yokoffing/filterlists/pull/39/commits/5d980ece1d4c936c6108acd848941e49f7c7b981. Here's what was left:

```
||asia^$doc
||beauty^$doc
||cn^$doc
||degree^$doc
||fit^$doc
||fyi^$doc
||garden^$doc
||live^$doc,domain=~marcello.live|~notgoogle.live
||quest^$doc
||su^$doc,domain=~kaihat.su
||shop^$doc,domain=~nsverify.shop
||surf^$doc
||zone^$doc
```

Note the exceptions for `live, su,` and `shop`. May not include those for DNS-level blocking.

We could use this as a baseline for TLD blocking, then introduce other metrics/research to justify what other TLDs should be blocked.

Again, the filterlist doesn't include [Dandelion's list](https://github.com/DandelionSprout/adfilt/blob/cc262d248387fe9d0b3b15bfdb5f9b2f1f40c7b4/Dandelion%20Sprout's%20Anti-Malware%20List.txt#L9-L38). Some of his fall under 10%. But he has researched beyond this basic rubric I've imposed, so I'd keep his entries (as a filterlist). To hardblock at the DNS level, the ones that don't have a bunch of exceptions are:

```
||agency^$doc,domain=~battlefield.agency|~shortcut.agency
||bid^$doc
||cfd^$doc
||discount^$doc
||gdn^$doc
||loan^$doc
||ooo^$doc
||sbs^$doc
```

Note the exceptions for `agency`. May not include those for DNS-level blocking.

Here's the merged list:

```
||agency^$doc,domain=~battlefield.agency|~shortcut.agency
||asia^$doc
||beauty^$doc
||bid^$doc
||cfd^$doc
||cn^$doc
||degree^$doc
||discount^$doc
||fit^$doc
||fyi^$doc
||garden^$doc
||gdn^$doc
||live^$doc,domain=~marcello.live|~notgoogle.live
||loan^$doc
||ooo^$doc
||quest^$doc
||sbs^$doc
||shop^$doc,domain=~nsverify.shop
||su^$doc,domain=~kaihat.su
||surf^$doc
||zone^$doc
```

**Comment by devipasigner** [comment]
> @devipasigner
> 
> > .email
> > .sex
> > .sexy
> > .recipes
> > .xxx
> > .yandex
> > .zone
> > .software
> 
> Spamhaus is not everything, but one tool to take into consideration:
> 
> **TLD**	**Spamhaus % Bad Domains**
> com (reference)	1.7%
> org (reference)	1.2%
> email	4%
> sex	0%
> sexy	0%
> recipes	0%
> xxx	0%
> yandex	0%
> zone	11.1%
> software	5.8%

Awesome,  there are a few entries that are constantly being used to spam on emails with phishing that need to be added though. I will gather them and send to review

**Comment by hagezi** [comment]
@yokoffing Thanks for your work. Exceptions cannot be handled in DNS rules. I would then have to free the corresponding domain with all with all subdomains against.

**Comment by hagezi** [comment]
Maybe we should take the list of @yokoffing as a common list. I would then parse it and convert it to `|*.tld^` to fit the rules for DNS. However, I would not like to lock out whole countries like `cn` and `ru`.

**Comment by yokoffing** [comment]
> Exceptions cannot be handled in DNS rules. I would then have to unblock the corresponding domain with all subdomains.

@hagezi Precisely. We need to be very selective as to what get blocked at the DNS-level.

A filterlist can provide a warning before navigation _and_ still allow navigation in uBlock Origin (not sure about AdGuard); hard-blocking at the DNS-level can't. User-friendliness dictates that we are more relaxed at the DNS level (your list) and stricter at adblock/filterlist level (my filterlist + Dandelion's malware list). So, as far as those four TLDs with exceptions/false positives in the combined list, I wouldn't include at all at the DNS level, personally. But that's your call.

> I would not like to lock out whole countries like `cn` and `ru`.

They should be removed at the DNS-level. Users can add to their personal list if they want to block it.

- We haven't mentioned it but `asia` is in the combined list and is used by very few sites. 10.8% bad. I'll keep that one.
- `ru` isn't in the combined list? 
- `cn` on my filterlist. I haven't had any complaints so far, but I will allowlist it for now.
- I'll separate these TLDs from the rest https://github.com/yokoffing/filterlists/pull/39/commits/8a6311669899b8e7a20555cb9889aafebf2714ed

**Comment by devipasigner** [comment]
Went through my email with over 500 pages of spam and the most abused by far are:
```
.ml
.sbs
.cf
.site
.gq
.ga
.tk
.top
.in
```

**Comment by yokoffing** [comment]
@devipasigner Thanks for doing that!

### Good
`.sbs` is already included in our combined list above.

### Too many false positives
- `.top` Too many false positives and probably more: https://github.com/DandelionSprout/adfilt/blob/2cbcfff6e62f6fff817a64390810188ab8903c08/Dandelion%20Sprout's%20Anti-Malware%20List.txt#L24
- same for `.ml` https://github.com/DandelionSprout/adfilt/blob/2cbcfff6e62f6fff817a64390810188ab8903c08/Dandelion%20Sprout's%20Anti-Malware%20List.txt#L15-L16
- and `.tk` https://github.com/DandelionSprout/adfilt/blob/2cbcfff6e62f6fff817a64390810188ab8903c08/Dandelion%20Sprout's%20Anti-Malware%20List.txt#L11-L12

### Possibly appropriate for soft-blocking by filter list, but not hard-blocking via DNS:
- `.online` One false positive noted but there may be others after we research further. I do recall there being a phishing used with this TLD last year. It is used legitimately for manga and video streaming sites, so I wouldn't block it entirely.
- `.fun` also has legitimate uses https://github.com/yokoffing/filterlists/blob/8cd755ed08bd300222b37d6f8b71b492fe6eae36/enhanced_site_protection.txt#L25
- `.cf`, `.ga`, `.gq` are on my hardened NextDNS list and they break third-party video streaming sites and have other unintended breakage (subrequests on pages). https://github.com/DandelionSprout/adfilt/blob/2cbcfff6e62f6fff817a64390810188ab8903c08/Dandelion%20Sprout's%20Anti-Malware%20List.txt#L14-L20

### Probably not appropriate but unsure
- `.site` has legitimate uses (Google, Fmovies, sports, etc.)
- `.in` is the domain for India. @hagezi doesn't want to block country domains

**Comment by Dynasty-Dev** [comment]
Hello all! Just tuning in here to share some ideas, perhaps there could be 2 versions a strict and more balanced list, kind of like the system @yokoffing already has going on, but perhaps more optimized. Why? Sure there may be some false positives caused by these tlds but thats the nature of tld and regex blocking.. of course we should still offer a balanced one. But some of the ones that have a few false positives are sites not many people will probably ever go on but will block a lot of the phishing sites. More good than bad.

**Comment by Dynasty-Dev** [comment]
> @devipasigner Thanks for doing that!
> ### Good
> 
> `.sbs` is already included in our combined list above.
> ### Too many false positives
> 
>     * `.top` Too many false positives and probably more: https://github.com/DandelionSprout/adfilt/blob/2cbcfff6e62f6fff817a64390810188ab8903c08/Dandelion%20Sprout's%20Anti-Malware%20List.txt#L24
> 
>     * same for `.ml` https://github.com/DandelionSprout/adfilt/blob/2cbcfff6e62f6fff817a64390810188ab8903c08/Dandelion%20Sprout's%20Anti-Malware%20List.txt#L15-L16
> 
>     * and `.tk` https://github.com/DandelionSprout/adfilt/blob/2cbcfff6e62f6fff817a64390810188ab8903c08/Dandelion%20Sprout's%20Anti-Malware%20List.txt#L11-L12
> 
> 
> ### Possibly appropriate for soft-blocking by filter list, but not hard-blocking via DNS:
> 
>     * `.online` One false positive noted but there may be others after we research further. I do recall there being a phishing used with this TLD last year. It is used legitimately for manga and video streaming sites, so I wouldn't block it entirely.
> 
>     * `.fun` also has legitimate uses https://github.com/yokoffing/filterlists/blob/8cd755ed08bd300222b37d6f8b71b492fe6eae36/enhanced_site_protection.txt#L25
> 
>     * `.cf`, `.ga`, `.gq` are on my hardened NextDNS list and they break third-party video streaming sites and have other unintended breakage (subrequests on pages). https://github.com/DandelionSprout/adfilt/blob/2cbcfff6e62f6fff817a64390810188ab8903c08/Dandelion%20Sprout's%20Anti-Malware%20List.txt#L14-L20
> 
> 
> ### Probably not appropriate but unsure
> 
>     * `.site` has legitimate uses (Google, Fmovies, sports, etc.)
> 
>     * `.in` is the domain for India. @hagezi doesn't want to block country domains
> 
> 
> Let me know your thoughts.
> 
> I moved `.online` `.fun` back to my filterlist to soft-block, and added `.in` as a new entry. https://github.com/yokoffing/filterlists/blob/0cf52963af4ae322771d30620bb3542b1c7813c2/enhanced_site_protection.txt
> 
> `.cf`, `.ga`, `.gq` are blocked by Dandelion's Malware list and have exceptions already there. I wouldn't block these at DNS-level either, though.

Hey! I'm just joining in , but  from what I see a lot of those listed for only soft blocking have very minimal false positives. Lots of those are used for malware redirects especially in emails and streaming sites and I think it is needed. Now like I wrote above, the 2 solutions would be: 1. Make 2 different lists, one light, one 'aggressive' but I wouldn't even call it that. 2. Make one big list with false positives constantly whitelisted using top lists and dandelions small selection already made

**Comment by devipasigner** [comment]
> @devipasigner Thanks for doing that!
> ### Good
> 
> `.sbs` is already included in our combined list above.
> ### Too many false positives
> 
>     * `.top` Too many false positives and probably more: https://github.com/DandelionSprout/adfilt/blob/2cbcfff6e62f6fff817a64390810188ab8903c08/Dandelion%20Sprout's%20Anti-Malware%20List.txt#L24
> 
>     * same for `.ml` https://github.com/DandelionSprout/adfilt/blob/2cbcfff6e62f6fff817a64390810188ab8903c08/Dandelion%20Sprout's%20Anti-Malware%20List.txt#L15-L16
> 
>     * and `.tk` https://github.com/DandelionSprout/adfilt/blob/2cbcfff6e62f6fff817a64390810188ab8903c08/Dandelion%20Sprout's%20Anti-Malware%20List.txt#L11-L12
> 
> 
> ### Possibly appropriate for soft-blocking by filter list, but not hard-blocking via DNS:
> 
>     * `.online` One false positive noted but there may be others after we research further. I do recall there being a phishing used with this TLD last year. It is used legitimately for manga and video streaming sites, so I wouldn't block it entirely.
> 
>     * `.fun` also has legitimate uses https://github.com/yokoffing/filterlists/blob/8cd755ed08bd300222b37d6f8b71b492fe6eae36/enhanced_site_protection.txt#L25
> 
>     * `.cf`, `.ga`, `.gq` are on my hardened NextDNS list and they break third-party video streaming sites and have other unintended breakage (subrequests on pages). https://github.com/DandelionSprout/adfilt/blob/2cbcfff6e62f6fff817a64390810188ab8903c08/Dandelion%20Sprout's%20Anti-Malware%20List.txt#L14-L20

@yokoffing thank you for the feedback, personally I think these entries are very much needed despite having a couple of false positives. There are millions of phishing domains using these TLD's and only a couple of legitimate that a regular user will ever visit.
Let me know what you think, I simply think the good extremely outweights the bad. @Dynasty-Dev proposed are pretty good solution however I don't think they would be too far from each other in terms of false positives/aggressiveness.

**Comment by yokoffing** [comment]
> 1. Make 2 different lists, one light, one 'aggressive' but I wouldn't even call it that. 2. Make one big list with false positives constantly whitelisted using top lists and dandelions small selection already made

@Dynasty-Dev That's already happening at the filterlist level. But this cannot be done at the DNS-level. You have to manually allowlist all the false positives -- which is fine for a personal setup, but impractical when you're making a list for many people. You're going to run into breakage.

> personally I think these entries are very much needed despite having a couple of false positives. There are millions of phishing domains using these TLD's and only a couple of legitimate domains using these TLDs that a regular user will ever visit.

@devipasigner I just want to avoid a scenario where someone is having to allowlist something every week. I may make some concessions and get feedback from users over time. Let me think on it.

Adblocking side isn't too bad with uBlock Origin since you can bypass the block and report the false positive later. It's DNS blocking that can be very frustrating.

**Comment by hagezi** [comment]
So friends, now you've hung up on me - I'm old! :D
Before I now no longer know what remains on the TLD list and what not, I'll sleep a night over it. :)

**Comment by Dynasty-Dev** [comment]
> > 1. Make 2 different lists, one light, one 'aggressive' but I wouldn't even call it that. 2. Make one big list with false positives constantly whitelisted using top lists and dandelions small selection already made
> 
> @Dynasty-Dev That's already happening at the filterlist level. But this cannot be done at the DNS-level. You have to manually allowlist all the false positives -- which is fine for a personal setup, but impractical when you're making a list for many people. You're going to run into breakage.
> 
> > personally I think these entries are very much needed despite having a couple of false positives. There are millions of phishing domains using these TLD's and only a couple of legitimate domains using these TLDs that a regular user will ever visit.
> 
> @devipasigner I just want to avoid a scenario where someone is having to allowlist something every week. I may make some concessions and get feedback from users over time. Let me think on it.
> 
> Adblocking side isn't too bad with uBlock Origin since you can bypass the block and report the false positive later. It's DNS blocking that can be very frustrating.

I understand! And I respect and love all the work you've done. Almost every NextDNS user has a well balanced profile because of you. (i personally use it too). However I think the amount of false positives from these TLDs is being very exaggerated. I personally use Hagezi's tld blocklist plus my own entries for 2 different households and I haven't gotten any false positive reports from them (only from blocklists). I personally use a lot of those free streaming sites for movies and sports (nba, football) too. As for using browser extensions, I agree, they should be applied everywhere however there are censorious where it won't help (for example, email apps on mobile (huge source of phishing and spam). Now the average techie will know its a scam, however the same can't be said about kids, wifes, grandparents

**Comment by yokoffing** [comment]
> I think theres just way too many malware and phishing domains using these TLDs to allow it, even if it means whitelisting 1/2 domains.

@devipasigner That sounds amazing --- until you're the list maintainer 🙃 

> Went through my emails with over 500 pages of spam and phishing and the most abused by far are:
> 
> ```
> .ml
> .sbs
> .cf
> .site
> .online
> .gq
> .ga
> .tk
> .top
> .in
> .fun (a bunch of nsfw phishing redirects from google docs pdf)
> ```

We'll give it a go.

 I have added/restored the ones listed above that not already in Dandelion's list, with what exceptions that I'm aware of https://github.com/yokoffing/filterlists/pull/39/commits/168e83d6fe1797d693f582beb600d83348f435b9

 **Edit:** Accidentally added `.top` and now removed, since Dandelion actively covers that https://github.com/yokoffing/filterlists/pull/39/commits/a02e70dc0a0317514271384f3c5c5947cf69a51b

This will block top-site navigations and not break sub-requests. Unlike blocking these at the DNS level, I don't anticipate too many false positives.

***

When we have the space to discuss [the rest of these domains](https://github.com/yokoffing/filterlists/blob/a02e70dc0a0317514271384f3c5c5947cf69a51b/enhanced_site_protection.txt#L42-L67), I'll push the [pull request](https://github.com/yokoffing/filterlists/pull/39) through. Keep an eye on it until then.


**Pull Request:** https://github.com/yokoffing/filterlists/pull/39
**yokoffing Enhanced Protection List:** https://raw.githubusercontent.com/yokoffing/filterlists/main/enhanced_site_protection.txt
**Dandelion Sprout's Anti-Malware List**: https://raw.githubusercontent.com/DandelionSprout/adfilt/master/Dandelion%20Sprout's%20Anti-Malware%20List.txt

**Comment by bestplayerbot** [comment]
found a false postive for *.fun yesterday

**Comment by hagezi** [comment]
So, I might do the following with my list:

I will take the top 10 Spamhaus TLDs as before, meaning every TLD that has been in the top 10 since the list was created will end up on the list. Exceptions below.

I additionally take over the TLDs from yokoffing by parsing his list. I will implement the exceptions with the `denyallow` modifier, example: `|*.agency^$denyallow=battlefield.agency|shortcut.agency`

Furthermore TLDs can be added manually of course.

I will exclude the following TLDs from the overall list:

Country specific TLDs, like:
```
cn
ru
in
co
uk
de
```

Other TLDs:
```
info
com
net
org
io
me
xyz
```

What do you think about such a solution?

**Comment by yokoffing** [comment]
Looks good. Here's what the 'combined list' looks like now (mine + Dande + @devipasigner's research on what should be restored), with comments. Please review: https://github.com/yokoffing/filterlists/blob/cd71a3ce16cad7717880394d2fa9e7d41711aa26/enhanced_site_protection.txt#L9-L62

@hagezi Now, here's the combined list with:
1) comment lines removed
2) `cn` `ru` `in` removed
3) alphabatized

```
||agency^$doc,domain=~battlefield.agency|~shortcut.agency
||asia^$doc
||beauty^$doc
||bid^$doc
||cf^$doc,domain=~google.cf|~rths.cf|~voitures.cf|~assembleenationale-rca.cf|~cps-rca.cf|~acap.cf|~miraculousladybug.cf|~scrat.cf
||cfd^$doc
||degree^$doc
||discount^$doc
||fit^$doc
||fun^$doc,domain=~libgen.fun|~gaggle.fun|~neal.fun|~bestgore.fun
||fyi^$doc
||ga^$doc,domain=~google.ga|~filtri-dns.ga|~dgdi.ga|~voitures.ga|~economie-gabon.ga|~9191.ga|~animevsub.ga
||garden^$doc
||gdn^$doc
||gq^$doc,domain=~deimos.gq|~inege.gq|~tvgelive.gq|~comprarcarros.gq
||live^$doc,domain=~marcello.live
||loan^$doc
||ml^$doc,domain=~google.ml|~mobili.ml|~melody.ml|~dcod.ml|~info-matin.ml|~amap.ml|~mastodon.ml|~worproject.ml|~nothingprivate.ml|~lingva.ml|~lemmy.ml|~bittor.ml|~noic.ml|~beatbump.ml|~gymlibrary.ml|~animevsub.ml|~prompt.ml|~biblioreads.ml
||monster^$doc,domain=~egybest.monster|~yts.monster|~cloudcdn.monster|~fedi.monster
||online^$doc,domain=~ero-labs.online|~amedia.online|~allhen.online|~chainsaw-man-manga.online|~manga1st.online
||ooo^$doc
||pw^$doc,domain=~libgen.pw|~petridish.pw|~palaugov.pw|~dpc.pw|~zikrap.pw|~demonoid.pw|~bittor.pw|~buttercup.pw|~rezka.pw|~darkcrystal.pw|~xor.pw|~fullhdfilmizlesene.pw|~b00k.pw|~gopass.pw|~vost.pw
||quest^$doc
||sbs^$doc
||shop^$doc,domain=~nsverify.shop
||site^$doc,domain=~business.site|~anitube.site|~wuxiaworld.site|~notube.site|~fmoviesto.site|~secreto.site|~metrolagu.site|~betphoenix.site|~cdmstudy.site
||su^$doc,domain=~kaihat.su
||surf^$doc
||tk^$doc,domain=~coolcmd.tk|~budterence.tk|~google.tk|~transportnews.tk|~c0d3c.tk|~anonytext.tk|~tokelau-info.tk|~fakaofo.tk|~loljp-wiki.tk|~ninetail.tk|~goshujin.tk|~graph.tk|~dls2.pokeacer.tk|~nolfrevival.tk|~coppersurfer.tk|~restricted-functions.tk|~bstweaker.tk|~nbd-media.tk|~glypx-pakhsh-nakon.tk|~gotofap.tk|~somepythonthings.tk
||top^$doc,domain=~corriente.top|~gdtot.top|~nicenature.top|~reminder.top|~magocoro.top|~castlevania.top|~suiten.top|~shucks.top|~1stream.top|~ambr.top|~techblog.top|~changlam10.top|~changlam11.top|~pdcdn1.top|~mastodon.top|~pressplay.top
||zone^$doc
```

**Comment by hagezi** [comment]
Thanks @yokoffing, looks good!

**Comment by hagezi** [comment]
DNS Syntax AdGuard Home, add to custom filtering rules for testing:

```
|*.agency^$denyallow=battlefield.agency|shortcut.agency
|*.asia^
|*.beauty^
|*.bid^
|*.cf^$denyallow=google.cf|rths.cf|voitures.cf|assembleenationale-rca.cf|cps-rca.cf|acap.cf|miraculousladybug.cf|scrat.cf
|*.cfd^
|*.degree^
|*.discount^
|*.fit^
|*.fun^$denyallow=libgen.fun|gaggle.fun|neal.fun|bestgore.fun
|*.fyi^
|*.ga^$denyallow=google.ga|filtri-dns.ga|dgdi.ga|voitures.ga|economie-gabon.ga|9191.ga|animevsub.ga
|*.garden^
|*.gdn^
|*.gq^$denyallow=deimos.gq|inege.gq|tvgelive.gq|comprarcarros.gq
|*.live^$denyallow=marcello.live
|*.loan^
|*.ml^$denyallow=google.ml|mobili.ml|melody.ml|dcod.ml|info-matin.ml|amap.ml|mastodon.ml|worproject.ml|nothingprivate.ml|lingva.ml|lemmy.ml|bittor.ml|noic.ml|beatbump.ml|gymlibrary.ml|animevsub.ml|prompt.ml|biblioreads.ml
|*.monster^$denyallow=egybest.monster|yts.monster|cloudcdn.monster|fedi.monster
|*.online^$denyallow=ero-labs.online|amedia.online|allhen.online|chainsaw-man-manga.online|manga1st.online
|*.ooo^
|*.pw^$denyallow=libgen.pw|petridish.pw|palaugov.pw|dpc.pw|zikrap.pw|demonoid.pw|bittor.pw|buttercup.pw|rezka.pw|darkcrystal.pw|xor.pw|fullhdfilmizlesene.pw|b00k.pw|gopass.pw|vost.pw
|*.quest^
|*.sbs^
|*.shop^$denyallow=nsverify.shop
|*.site^$denyallow=business.site|anitube.site|wuxiaworld.site|notube.site|fmoviesto.site|secreto.site|metrolagu.site|betphoenix.site|cdmstudy.site
|*.su^$denyallow=kaihat.su
|*.surf^
|*.tk^$denyallow=coolcmd.tk|budterence.tk|google.tk|transportnews.tk|c0d3c.tk|anonytext.tk|tokelau-info.tk|fakaofo.tk|loljp-wiki.tk|ninetail.tk|goshujin.tk|graph.tk|dls2.pokeacer.tk|nolfrevival.tk|coppersurfer.tk|restricted-functions.tk|bstweaker.tk|nbd-media.tk|glypx-pakhsh-nakon.tk|gotofap.tk|somepythonthings.tk
|*.top^$denyallow=corriente.top|gdtot.top|nicenature.top|reminder.top|magocoro.top|castlevania.top|suiten.top|shucks.top|1stream.top|ambr.top|techblog.top|changlam10.top|changlam11.top|pdcdn1.top|mastodon.top|pressplay.top
|*.zone^
```

@yokoffing new Exception for `pw` : `core.pw`

**Comment by yokoffing** [comment]
> Exception for `pw` : `core.pw`

@hagezi Thanks for the heads up. What is that? I can't get the site to load.

[`.pw` is in Dande's list](https://github.com/DandelionSprout/adfilt/blob/4a90c37a43d86c0f768a7b658dd1eccf523c7309/Dandelion%20Sprout's%20Anti-Malware%20List.txt#L22). Open a pull request with him.

**Comment by yokoffing** [comment]
How do we feel about restoring `.loans` (plural) to the list since we're already blocking `.loan` (singular)?

**Comment by hagezi** [comment]
> @hagezi Thanks for the heads up. What is that? I can't get the site to load.

Needed for Wargaming Games

https://myip.ms/view/hosts/2920936/core_pw.html

**Comment by yokoffing** [comment]
I've further refined the `Spamhaus` section. Again, I used the 10% as a starting point. It's not Bible. And Spamhaus is not the "be all, end all" of the cybersecurity world, but I like to reference their metrics. Lastly, note that this isn't touching the TLDs provided by Dandelion and @devipasigner.

I took into account whether these TLDs receive low or high traffic and whether the TLDs are used by few or many sites. I then visited popular (if any were in the top 100k or 1M sites) and random sites to see firsthand if we should bother to keep them.

Total of 3 or 4 = stay
Total of 1 or 2 = remove

| **TLD** | **Total** | **Spamhaus 10 Most Abused TLDs** | **Used by low-traffic or low-quality sites** | **Few Domains with TLD in existence** | **Many scammy rando sites (random searches)** |
|:-------:|:---------:|:--------------------------------:|:--------------------------------------------:|:-------------------------------------:|:---------------------------------------------:|
|   asia  |     0     |                                  |                                              |                                       |                                               |
|  beauty |     4     |                 x                |                       x                      |                   x                   |                       x                       |
|  degree |     3     |                                  |                       x                      |                   x                   |                       x                       |
|   fit   |     3     |                 x                |                       x                      |                   x                   |                                               |
|   fyi   |     1     |                                  |                                             |                   x                   |                                               |
|  garden |     1     |                                  |                                              |                   x                   |                                               |
|   live  |     1     |                 x                |                                              |                                       |                                               |
|  quest  |     3     |                                  |                       x                      |                   x                   |                       x                       |
|   shop  |     1     |                                  |                       x                      |                                       |                                               |
|    su   |     1     |                                  |                       x                      |                                       |                                               |
|   surf  |     2     |                 x                |                                              |                   x                   |                                               |
|   zone  |     3     |                                  |                       x                      |                   x                   |                       x                       |

- `.surf` is an exception to the rule: Spamhaus has it at 326/421 bad domains, so I'll take their word for it for now.
- `.garden` was only 0.9% bad on Spamhaus, and every website I checked looked fine. I must have missed that one in the original purge.
- `.shop` was only 8.9% bad on Spamhaus. I must have also missed that one in the original purge.

@hagezi Therefore, we can remove the following:

```
asia
fyi
garden
live
shop
su
```

***

**Comment by bestplayerbot** [comment]
why is *.kp blocked it's the TLD for north korea

**Comment by hagezi** [comment]
This will no longer be the case in the new list. When @yokoffing is finished with his list, I will parse it daily and convert it into DNS format. I will exclude country-specific TLDs. I will also not add any entries manually, yokoffing's list will be the only basis.

**Comment by hagezi** [comment]
New list is live; https://raw.githubusercontent.com/hagezi/dns-blocklists/main/adblock/spam-tlds.txt

Merged lists: Spamhaus Top10, @yokoffing @DandelionSprout

Excluded TLDs: `.cn` `.ru` `.in`

Please check ...

**Comment by iam-py-test** [comment]
Yay, thanks!

**Comment by yokoffing** [comment]
@hagezi Run the script again. I just pushed a pull request from yesterday that comments out TLDs we haven't discussed (e.g., `.icu`, and to @iam-py-test's point, `.xxx`). The version you have did not do this.

This will provide us a nice core, foundation list 😄

**Comment by yokoffing** [comment]
@hagezi `.yandex` is on there. Just found that one and removed from my list. That was my bad. Otherwise, the list is beautiful!

@iam-py-test Thanks for joining in! I've been wanting to ask you: Can you explain more on why you block `.win` and `.cricket` in [your list](https://github.com/iam-py-test/my_filters_001/blob/main/enhanced_protection.txt)? Thank you ahead of time for your insight.

**Comment by hagezi** [comment]
Thanks @yokoffing, updated. Thanks for joining @iam-py-test!

**Comment by yokoffing** [comment]
@hagezi I know syntax will change based on format of your list (e..g, domain, hosts, etc.), but for the Adblock list, what is the reasoning to change from `||tld^$doc,domain=` to `|*.tld^$denyallow=`? The latter can't be recognized by uBlock Origin.

![Screenshot 2023-01-06 104929](https://user-images.githubusercontent.com/11689349/211048427-9f2b6ec9-9665-4b28-af2c-e2cc3970ad6c.png)

Will the format you've shared work at the DNS level (AdGuard Home, NextDNS, etc.)?

**Comment by iam-py-test** [comment]
> @iam-py-test Thanks for joining in! I've been wanting to ask you: Can you explain more on why you block .win and .cricket in [your list](https://github.com/iam-py-test/my_filters_001/blob/main/enhanced_protection.txt)? Thank you ahead of time for your insight.

Honestly, that list is very poorly thought out. I just copied a ton of random TLDs from somewhere else (I don't remember where), and then over time removed some that had known FPs reported to other lists. I have put [pretty much 0 work](https://github.com/iam-py-test/my_filters_001/commits/main/enhanced_protection.txt) into that list, so I wouldn't put much creditably into it.

**Comment by iam-py-test** [comment]
I think that might be because $denyallow requires $domain (https://github.com/DandelionSprout/adfilt/blob/master/Wiki/SyntaxMeaningsThatAreActuallyHumanReadable.md#blocking-1)

**Comment by hagezi** [comment]
@iam-py-test denyallow should also work with the normal syntax `||TLD^`:
https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists#denyallow

**Comment by iam-py-test** [comment]
> @iam-py-test denyallow should also work with the normal syntax `||TLD^`: https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists#denyallow

Not in uBlock Origin: https://github.com/gorhill/uBlock/wiki/Static-filter-syntax#denyallow
![image](https://user-images.githubusercontent.com/84232764/211051996-37235a81-b1ee-4267-8820-2855efec2cc6.png)

**Comment by DandelionSprout** [comment]
`$denyallow` in uBO requires being coupled with `$domain`, for some reason.

**Comment by hagezi** [comment]
Thanks @DandelionSprout, This will then no longer work with AdGuard Home, or will it?

**Comment by gorhill** [comment]
The reason is that a long time ago there was a commit with `denyallow` mistakenly without a `domain` option and this broke the whole internet for uBO users.

**Comment by hagezi** [comment]
@yokoffing https://raw.githubusercontent.com/hagezi/dns-blocklists/main/adblock/spam-tlds-ublock.txt

**Comment by hagezi** [comment]
@gorhill thanks for the clarification.

**Comment by yokoffing** [comment]
> `https://raw.githubusercontent.com/hagezi/dns-blocklists/main/adblock/spam-tlds-ublock.txt`

Works in uBO!

@hagezi Check your exclusions / exception lines. `.cn .ru .in` were reintroduced.

**Comment by hagezi** [comment]
@yokoffing only for uBlock Version.

**Comment by yokoffing** [comment]
> only for uBlock version, not for DNS version.

Very nice! Everything looks great, then.

**Comment by hagezi** [comment]
Thanks for the great cooperation to all involved, that's how it has to be. I'll leave the issue open, there are still a few things to clarify and then you have a point of contact.

**Comment by hagezi** [comment]
The parsed data from the individual lists is in converted form in my Data Collection if anyone wants to track changes to the individual sources:

https://github.com/hagezi/dns-data-collection/commits/main/data/TLDs_spamhaus.txt
https://github.com/hagezi/dns-data-collection/commits/main/data/TLDs_yokoffing.txt
https://github.com/hagezi/dns-data-collection/commits/main/data/TLDs_dandelionsprout.txt

**Comment by hagezi** [comment]
@yokoffing 
Found an exception for `CN`:
`nvgs.nvidia.cn` is required for the GEFORCE Experience login. This is called `accounts.nvgs.nvidia.cn`.

**Comment by yokoffing** [comment]
@hagezi https://github.com/yokoffing/filterlists/pull/42/commits/7c16d97f94ecbe887bb1b9186c68f2f0f07cecd7

**Comment by devipasigner** [comment]
@hagezi .site false positive, fitgirl-repacks.site

**Comment by hagezi** [comment]
Ping source: @yokoffing

**Comment by yokoffing** [comment]
So, some updates for my [filterlist](https://github.com/yokoffing/filterlists/blob/main/enhanced_site_protection.txt):
* @bestplayerbot has added a ton of exceptions for `.fun` and `.ru` (thanks!). But I think it raises a larger concern of high false positives from blocking **country-specific** TLDs (which is why @hagezi allowlists these for the DNS version).
* I think many of these should be manually enabled by users who don't live in those countries.
   * e.g., `.in` "[The filter list] blocks all banking websites in India and popular sites like `amazon.in` etc." https://github.com/yokoffing/filterlists/pull/59/files
* Also, to speak bluntly, these are too much work to allowlist. I have other (unpaid!) projects to maintain on top of school and work.
 * Therefore, I will be allowlisting `.cn`, `.in`, and `.ru` in the next major update.
    * If you want to retain them as a filterlist, add them to your custom rules for uBO/AdGuard.
    * If someone wants to create a "block country-specific TLDs" list and use my rules, they're welcome to do so.

Thank you for coming to my TED Talk

**Comment by hagezi** [comment]
Thanks @yokoffing, I absolutely agree.

**Comment by iam-py-test** [comment]
> If someone wants to create a "block country-specific TLDs" list and use my rules, they're welcome to do so.

I have copied a few of your entries, added some new domains to the allowlist, and uploaded them here: https://github.com/iam-py-test/my_filters_001/tree/main/region_blocklist
So far I have in and ru, but I might add a few more.

**Comment by bestplayerbot** [comment]
.cn is still there
![image](https://user-images.githubusercontent.com/96731476/213839637-9e22f87a-47b5-4d86-bfa5-c01ee0f2d925.png)

**Comment by hagezi** [comment]
> .cn is still there ![image](https://user-images.githubusercontent.com/96731476/213839637-9e22f87a-47b5-4d86-bfa5-c01ee0f2d925.png)

Thanksm fixed. Came via spamHAUS.

**Comment by RicoHeat** [comment]
Just my 2 cents on the matter..
https://www.reddit.com/r/uBlockOrigin/comments/10s4n1a/how_to_combine_multiple_tlds_into_one/

Excellent work everyone!!! 👍

**Comment by hagezi** [comment]
Thanks @RicoHeat

**Comment by krystian3w** [comment]
Why list have `||*.`?

**Comment by bestplayerbot** [comment]
![image](https://user-images.githubusercontent.com/96731476/219824788-9877f98f-df0a-4f21-8689-c5e8e0aa3d1a.png)
useful to find false positives

Allow
```
aclub.okinawa
activity.okinawa
aquadrop.okinawa
asatoyasuba.okinawa
beachboys.okinawa
churaumi.okinawa
dancestudioon.okinawa
diver.okinawa
ecco.okinawa
eight.okinawa
emoh.okinawa
fidelity.okinawa
growth.okinawa
gukuru.okinawa
holdbyhand.okinawa
jupiter.okinawa
kanoa.okinawa
kerouac.okinawa
kimishimagroup.okinawa
kokorononekko.okinawa
makeman.okinawa
mansaku.okinawa
minsyo.okinawa
morinaga.okinawa
naminouta.okinawa
nanaironote.okinawa
okmarine.okinawa
pacificview.okinawa
rentakun.okinawa
retime.okinawa
ryukyugobyou.okinawa
saya.okinawa
seek.okinawa
seitai.okinawa
selfstorage.okinawa
sgt.okinawa
shigaku.okinawa
simplelife.okinawa
sir.okinawa
sisei.okinawa
soilcure.okinawa
strawberry.okinawa
suba.okinawa
suncoast.okinawa
takakogyo.okinawa
thekucha.okinawa
tsukishiro.okinawa
yamabuki.okinawa
yorozu.okinawa
zwwa.okinawa
```

**Comment by hagezi** [comment]
@bestplayerbot Thanks, I have to adapt my script so that denyallow can be implemented for TLDs from the SpamHaus source. Have you checked the Okinawas to see what's behind them? They look wild ... :)

**Comment by bestplayerbot** [comment]
I visited all of them but I don't understand Japanese so I don't exactly know what they are.

**Comment by yokoffing** [comment]
Might need to use something with native translation support Chrome or Edge in a private window to determine the content of the sites.

**Comment by bestplayerbot** [comment]
Visited all of them again on chrome
`growth.okinawa` didn't load on chrome

**Comment by hagezi** [comment]
@bestplayerbot Thanks!
Implemented exclusions for the SpamHAUS TLDs. The okinawa's are in now.

**Comment by hagezi** [comment]
@bestplayerbot If you want to add exclusions for the [SpamHaus TLDs](https://github.com/hagezi/dns-data-collection/blob/main/data/TLDs_spamhaus.txt), just "pull request" [this file](https://github.com/hagezi/dns-data-collection/blob/main/data/TLDs_spamhaus-exclusion.txt). If new TLDs appear for which an exclusion is necessary, simply create a new line for the TLD. Only applies to TLDs that only come through SpamHaus.

**Comment by RicoHeat** [comment]
> I visited all of them but I don't understand Japanese so I don't exactly know what they are.

@Yuki2718

**Comment by Yuki2718** [comment]
Not scam, legit sites.

**Comment by hagezi** [comment]
Thanks @Yuki2718 ;)

**Comment by bestplayerbot** [comment]
Is *.okinawa going to get removed?

**Comment by hagezi** [comment]
@bestplayerbot No, why?
https://github.com/hagezi/dns-blocklists/blob/a2c2bbeb58225b8c9fe2647c783276411b23bfcc/adblock/spam-tlds.txt#L31

**Comment by bestplayerbot** [comment]
I just wanted to know if it is getting removed.

**Comment by Yuki2718** [comment]
> why?
> 

As it has nothing to do with scam/spam? There will be many more legitimate sites using `.okinawa`.

**Comment by hagezi** [comment]
@Yuki2718 https://www.spamhaus.org/statistics/tlds/
Rank 2 among the top spam TLDs ...

**Comment by Yuki2718** [comment]
Do they provide actual example? Frankly, I haven't ever felt SH is very reliable.

**Comment by bestplayerbot** [comment]
False Positive: `nic.*`
```
nic.cf
nic.ga
nic.gq
nic.ml
nic.tk
```
doesn't load

**Comment by hagezi** [comment]
`nic.*` What should that be?

**Comment by bestplayerbot** [comment]
```
nic.agency
nic.beauty
nic.bid
nic.cfd
nic.degree
nic.discount
nic.fit
nic.fun
nic.fyi
nic.gdn
nic.loan
nic.loans
nic.monster
nic.okinawa
nic.online
nic.ooo
nic.pw
nic.quest
nic.rest
nic.sbs
nic.site
nic.surf
nic.top
nic.uno
nic.zone
```

**Comment by hagezi** [comment]
So there would be exceptions for almost all spam TLDs. Does this really make sense for someone who uses a spam TLD list?

**Comment by bestplayerbot** [comment]
Every TLD should have a false positive.

**Comment by bestplayerbot** [comment]
> ```
> nic.agency
> nic.beauty
> nic.bid
> nic.cfd
> nic.degree
> nic.discount
> nic.fit
> nic.fun
> nic.fyi
> nic.gdn
> nic.loan
> nic.loans
> nic.monster
> nic.okinawa
> nic.online
> nic.ooo
> nic.pw
> nic.quest
> nic.rest
> nic.sbs
> nic.site
> nic.surf
> nic.top
> nic.uno
> nic.zone
> ```

When will these get allowed?

**Comment by hagezi** [comment]
Does this really make sense?

**Comment by xRuffKez** [comment]
Please whitelist domain api-cloudfront.life360.com or it breaks the App Life360

**Comment by bestplayerbotphone** [comment]
It's not possible for a TLD to be 100% malicious

False positive for `*.okinawa`
enokinawa.okinawa

*.okinawa should be removed because I didn't find a single malicious domain while searching `site:okinawa` in multiple search engines

**Comment by hagezi** [comment]
spamHAUS sees it differently, 69% bad domains, meanwhile on the index place 2:

https://www.spamhaus.org/statistics/tlds/

**Comment by bestplayerbot** [comment]
I think the TLDs added from spamhaus aren't getting removed when they are no longer on the top 10 list

**Comment by hagezi** [comment]
Yes, those that are on the top 10 list or have been on the top 10 list in the past will be added.

**Comment by bestplayerbot** [comment]
@hagezi the latest commit caused some problems

**Comment by hagezi** [comment]
@yokoffing @bestplayerbot 

Syntax changed in https://raw.githubusercontent.com/DandelionSprout/adfilt/master/Alternate%20versions%20Anti-Malware%20List/AntiMalwareAdGuardHome.txt

```
|*.pw^
@@||libgen.pw^
@@||petridish.pw^
@@||palaugov.pw^
@@||dpc.pw^
@@||buttercup.pw^
@@||rezka.pw^
@@||darkcrystal.pw^
@@||xor.pw^
@@||fullhdfilmizlesene.pw^
@@||gopass.pw^
@@||vost.pw^
@@||core.pw^
@@||bittor.pw^
```

**Comment by hagezi** [comment]
I have removed the TLDs for now, we should think about what to do with them. A complete unblocking of the exclusions with all subdomains by `@@||` I think is not a good idea. Probably the syntax was changed because the AdGuard host compiler can't handle `denyallow`.

**Comment by hagezi** [comment]
I have added the last version of the old syntax manually for now. So the list is complete again.

**Comment by hagezi** [comment]
Finally, i have adjusted my parsing in the script so that i can handle this kind of syntax. So the source of DandelionSprout is integrated again.

**Comment by bestplayerbot** [comment]
false positive for *.bar
`fap.bar` (NSFW)

**Comment by hagezi** [comment]
@yokoffing Exclusion for online: kinotickets.online

**Comment by devipasigner** [comment]
@hagezi Technitium can be added to the AdBlock supported section.

**Comment by bestplayerbot** [comment]
please add the exceptions for *.monster again

**Comment by hagezi** [comment]
@yokoffing switched from `,domain` to `,from`?

**Comment by yokoffing** [comment]
@bestplayerbot did that. Should I switch it back?

**Comment by hagezi** [comment]
> please add the exceptions for *.monster again

The monster TLD was commented out in the DS list, so it was taken from the SpamHAUS list without exclusions. I have taken over the exclusions from the last version.

**Comment by yokoffing** [comment]
> It can now process domain or from.

ok. Good. 

> What is the advantage of from?

There is none. It was made just for the sake of change. Which is not best practice.

**Comment by yokoffing** [comment]
@bestplayerbot I appreciate your many contributions to this project, but please be mindful of the unintended consequences of certain changes. This case illustrates why we should only make changes that are necessary and well-justified.

**Comment by bestplayerbot** [comment]
false positive: `digitalcitizen.life`

**Comment by devipasigner** [comment]
False Positive Report [.lol]: `nickchan.lol`, `cdn.nickchan.lol`.

**Comment by devipasigner** [comment]
> @devipasigner `lol` is not on the list.

Oh.. you're right I checked and it is blocked by a cname domain on your list, I'll make a separate issue for this.

**Comment by hagezi** [comment]
@devipasigner I have not found anything that blocks the domains.

not Blocked:
```
nickchan.lol
cdn.nickchan.lol - vmi1107677.contaboserver.net
```

**Comment by devipasigner** [comment]
> @devipasigner I have not found anything that blocks the domains.
> 
> not Blocked:
> 
> ```
> nickchan.lol
> cdn.nickchan.lol - vmi1107677.contaboserver.net
> ```

Yes, thats the exact domain thats blocked. `vmi1107677.contaboserver.net` blocked by `contaboserver.net` in your Ultimate list.

**Comment by bestplayerbot** [comment]
false positive: namu.wiki

**Comment by bestplayerbot** [comment]
the list didn't update on gitlab

**Comment by bestplayerbot** [comment]
why is there `||^`

**Comment by bestplayerbot** [comment]
@DandelionSprout 
false positive: `vipbj.monster` (NSFW)
same as
```
vipbj.link
vipbj.beauty
```

**Comment by hagezi** [comment]
> why is there `||^`

Seems to be a bug in my script, I'll have to look into it.

**Comment by mahtd** [comment]
Here's a legitimate `.biz`: `luckysbarbershop.biz`

**Comment by hagezi** [comment]
Thanks @mahtd, will be added to the `biz` exclusions.

**Comment by bestplayerbot** [comment]
Nitter instances
```
notabird.site
nitter.pw
```
@DandelionSprout @yokoffing

**Comment by bestplayerbot** [comment]
biblioreads.ml is dead it's replaced by https://biblioreads.eu.org
@yokoffing

**Comment by EthanBezz** [comment]
False Positive: `stepone.life`
Australian underwear store

False Positive: `circles.life`
Mobile plan provider, operating in Singapore, Taiwan, Australia, ect.

**Comment by hagezi** [comment]
Thanks @EthanBezz, fixed in the next update.

**Comment by yokoffing** [comment]
@hagezi `plooshiesaresocute.wtf` should be a `denyallow` for TLD `.wtf`.

**Comment by snekul** [comment]
Only false positive we've run across yet on this TLD blocklist:   loveinfaith.life
A Shopify-hosted Christian clothing store

Thanks for all your hard work on these block lists!

**Comment by hagezi** [comment]
@snekul Thanks, fixed.

**Comment by yokoffing** [comment]
@hagezi We can incorporate this [list](https://github.com/LennyFox/Blocklists/blob/master/Block_non_latin_TLDs.txt) to easily block non-latin TLDs.

**Comment by DandelionSprout** [comment]
I presume its 150 entries could be replaced with simply `/(://|^)[a-z0-9-.]{2,}\.xn--[a-z0-9]{4}($|/)/`

**Comment by BourbonCrow** [comment]
albion.zone  needs to be whitelisted 
reason: its the servers for the game Albion Online, if its blocked you cant see server status or patch or play

![image](https://github.com/hagezi/dns-blocklists/assets/6895650/78d262ef-bd1f-4fc1-b4f1-257fe22e1ed0)

**Comment by hagezi** [comment]
Thanks @BourbonCrow, ping @yokoffing for exclusion.

**Comment by yokoffing** [comment]
I'd like to reevaluate `.online` and `.fun`. The exceptions have become comically long, especially the former.

I [originally proposed](https://github.com/hagezi/dns-blocklists/issues/143#issuecomment-1371494253) that these two were appropriate for soft-blocking by filter list, but not hard-blocking via DNS.

**Comment by hagezi** [comment]
@yokoffing I see it that way too.

**Comment by DandelionSprout** [comment]
`.online` I feel is pretty safe and should potentially never have been considered for blocking in the first place.

`.fun` has at times been used in e.g. phone competition scams, but there are likely better ways to narrow those domains down.

**Comment by yokoffing** [comment]
@hagezi @DandelionSprout

Thanks! I'd also like feedback on `.zone` and `.site`, then I think I'm okay as far as pruning the list.

**Comment by hagezi** [comment]
@yokoffing : https://github.com/hagezi/dns-blocklists/issues/1265

**Comment by BourbonCrow** [comment]
![image](https://github.com/hagezi/dns-blocklists/assets/6895650/78fad1f6-bc9d-478e-a8ec-8987d742bbaa)
this is the weirdest thing ive seen idk why they have this but and not just use their .se domain..  astmaochallergilinjen.se is being blocked cause it needs the dns lookup of master-7rqtwti-fd5bjuy2l3c3u.eu-5.platformsh.site  its a website about asthma and allergys and should probably be allowed was something my grandma tried to access so not sure but looks legit

**Comment by iam-py-test** [comment]
Would you mind posting the actual domain, as it is hard to read from a screenshot?
I suspect the second domain (the .site one) is a [CNAME](https://en.wikipedia.org/wiki/CNAME_record) of the first (basically, going to the first domain means you actually go to the second)
Thanks

**Comment by BourbonCrow** [comment]
@iam-py-test i did? isnt the domain in the text i wrote?

**Comment by iam-py-test** [comment]
Oh, I'm a idiot 🤦‍♀️ 
Sorry

**Comment by BourbonCrow** [comment]
While we are at it slussen.biz needs a whitelist as well for TLD

**Comment by hagezi** [comment]
ping @yokoffing: exclusion for `platformsh.site`, see https://github.com/hagezi/dns-blocklists/issues/143#issuecomment-1656907164

@BourbonCrow `slussen.biz` will be excluded on next update.

**Comment by itsnebulalol** [comment]
Not sure if .win is still blocked, but if so, I know a couple legit sites that use .win. For example, `stkc.win` and `quickvids.win`. Worth adding them to an allowlist if keeping .win blocked.

**Comment by TPS** [comment]
Is still where to propose TLD exceptions?  If so, I'd like to propose whitelisting https://jp.foundation/, which is Jaques Pépin's Foundation for culinary education.

**Comment by hagezi** [comment]
@TPS yes, will be excluded.

**Comment by TPS** [comment]
@hagezi I'm sorry, I've 1 more: https://encode.su/ is a legitimate software compression  forum which moved from `.ru` domain awhile back.

**Comment by TPS** [comment]
https://userstyles.world/ is a legit site for userstyles.  May it be excluded, please?
